Models/Entity.cs

```diff
@@ -1,12 +1,15 @@
-﻿using System.ComponentModel.DataAnnotations;
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 
-namespace StoreManagement.Models
+namespace StoresManagement.Models
 {
     public class Entity
     {
         public int Id { get; set; }
 
         [Display(Name = "Entity Name")]
         public string Name { get; set; }
+
+        public virtual ICollection<Branch> Branches { get; set; }
```

Comment:
good behavior is to assign instance of HashSet<Branch> in constructor to avoid null reference exception. Tip: type `ctor` and press tab twice

Reply:
Should it be like below?

        public Entity()
        {
            Branches = new List<Branch>();
        }

Reply:
```public Entity()
    {
        Branches = new HashSet<Branch>();
    }```

read about HashSets - they work fast. Accessing element take more or less O(1). You can be sure (assumption) Branches in DbSet will be unique.

https://stackoverflow.com/questions/150750/hashset-vs-list-performance